File: sample-infrastructure-services-domain/dotnet/src/queries/queries.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Sample.Domain.Aggregates;
using Sample.Domain.Aggregates.Parent;
using Sample.Services.Queries;
using Sample.Services.Queries.Models;

namespace Sample.Infrastructure.Services.Domain.Queries 
{
    public class DomainQueries : IQueries
    {

        private readonly ISampleUnitOfWorkFactory _uowFactory;

        public DomainQueries(ISampleUnitOfWorkFactory uowFactory)
        {
            _uowFactory = uowFactory;
        }

        private static SomethingDto ToSomethingDto(Parent parent) => new SomethingDto()
        {
            SomeId = parent.Id,
            SomeData = parent.Description
        };

        private async Task<TResult> InRollbackTransactionAsync<TResult>(Func<ISampleUnitOfWork, Task<TResult>> asyncOperation)
        {
            using (var uow = _uowFactory.Create())
            {
                var transactionId = await uow.BeginAsync();
                return await asyncOperation(uow);
            }
        }

        public async Task<IEnumerable<SomethingDto>> GetSomethingsAsync() => await InRollbackTransactionAsync(async uow => 
        {
            var parents = await uow.Parents.GetAsync();
            return parents.Select(ToSomethingDto);
        });

        public async Task<SomethingDto> GetSomethingByIdAsync(int id) => await InRollbackTransactionAsync(async uow => 
        {
            var parent = await uow.Parents.GetAsync(id);
            return parent != null ? ToSomethingDto(parent) : null;
        });

    }
}

File: sample-infrastructure-services-domain/npm/src/queries/queries.ts
import { inject, injectable } from "inversify";
import { ISampleUnitOfWork, ISampleUnitOfWorkFactory, Parent, TYPES as Domain } from "sample-domain";
import { IQueries, ISomethingDto } from "sample-services";

@injectable()
export class DomainQueries implements IQueries {

    private readonly _uowFactory: ISampleUnitOfWorkFactory;

    constructor(@inject(Domain.Aggregates.ISampleUnitOfWorkFactory) uowFactory: ISampleUnitOfWorkFactory) {
        this._uowFactory = uowFactory;
    }

    private static toSomethingDto(parent: Parent): ISomethingDto {
        return {
            someId: parent.id,
            someData: parent.description
        };
    }

    private async inRollbackTransactionAsync<TResult>(asyncOperation: (uow: ISampleUnitOfWork) => Promise<TResult>): Promise<TResult> {
        let uow: ISampleUnitOfWork | null = null;
        let transactionId = "";
        let results: TResult;
        try {
            uow = this._uowFactory.create();
            transactionId = await uow.beginAsync();
            results = await asyncOperation(uow);
        }
        finally {
            if (uow) {
                transactionId && await uow.rollbackAsync();
                uow.dispose();
            }
        }
        return results;
    }

    async getSomethingsAsync(): Promise<ISomethingDto[]> {
        return await this.inRollbackTransactionAsync(async uow => {
            const parents = await uow.parents.getAsync();
            return parents.map(DomainQueries.toSomethingDto);
        });
    }    

    async getSomethingByIdAsync(id: number): Promise<ISomethingDto | null> {
        return await this.inRollbackTransactionAsync(async uow => {
            const parent = await uow.parents.getAsync(id);
            return parent ? DomainQueries.toSomethingDto(parent) : null;
        });
    }

}